app/src/main/java/com/jaredrummler/mobiletest/di/NetworkModule.kt

```diff
@@ -0,0 +1,68 @@
+package com.jaredrummler.mobiletest.di
+
+import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import com.jaredrummler.mobiletest.BuildConfig
+import com.jaredrummler.mobiletest.networking.ApiService
+import com.jaredrummler.mobiletest.networking.SeedApi
+import dagger.Module
+import dagger.Provides
+import io.reactivex.Scheduler
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+class NetworkModule {
+
+  @Provides
+  @Singleton
+  @Named(SCHEDULER_IO)
+  fun provideIoScheduler(): Scheduler = Schedulers.io()
+
+  @Provides
+  @Singleton
+  @Named(SCHEDULER_UI)
+  fun provideUiScheduler(): Scheduler = AndroidSchedulers.mainThread()
+
+  @Provides
+  @Singleton
+  fun provideOkHttpClient(): OkHttpClient {
+    val builder = OkHttpClient.Builder()
+    builder.connectTimeout(30, TimeUnit.SECONDS)
+    builder.readTimeout(30, TimeUnit.SECONDS)
+    builder.writeTimeout(30, TimeUnit.SECONDS)
+    if (BuildConfig.DEBUG) {
+      builder.addInterceptor(HttpLoggingInterceptor()
+          .setLevel(HttpLoggingInterceptor.Level.BODY))
+    }
+    return builder.build()
```

Comment:
`return OkHttpClient.Builder()
.connectTimeout(30, TimeUnit.SECONDS)
.readTimeout(30, TimeUnit.SECONDS)
.writeTimeout(30, TimeUnit.SECONDS)
.let{
if (BuildConfig.DEBUG) {
addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
    }
}
.build()`

Reply:
That would be more functional